src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,333 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http:/service-name")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "service-name",
+    RequestAcl(Http("^/service-name".r -> "/"))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http:/service-name]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "service-name"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/service-name"
+                  rewrite = "/"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+
+
+##### Services mapped with preserved path
+
+
+Given the following service declaration, the applications or services expect the context root to be preserved when passing through ConductR's proxy, then the service URI can be instead expressed as "http:/service-name?preservePath"
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http:/service-name?preservePath")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "service-name",
+    RequestAcl(Http("^/service-name".r))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http:/service-name?preservePath"]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "service-name"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/service-name"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+#### Supplying customised HAProxy template configuration
+
+The following services definition will require a customised HAProxy configuration.
+
+The [[Dynamic proxy configuration|DynamicProxyConfiguration]] describes the process of generating a customised HAProxy configuration.
+
+
+##### A note on system, system version, and endpoint label
```

Comment:
I'd remove this headline. The text of this headline can be part of the previous headline `#### Supplying customised HAProxy template configuration`.